packages/wrangler/src/module-collection.ts

```diff
@@ -101,6 +81,69 @@ export default function createModuleCollector(props: {
 	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
 	rulesToRemove.forEach((rule) => rules!.splice(rules!.indexOf(rule), 1));
 
+	return { rules, rulesToRemove };
+}
+
+export async function matchFiles(
+	files: string[],
+	rules: Config["rules"],
+	rulesToRemove: Config["rules"]
```

Comment:
NIT: these are not rules to remove. They have already been removed. We are passing in here rules that have been removed, purely so we can give warning messages if they would have matched a file.